### ISSUE TITLE
Crash in image-upload

### ISSUE THREAD

**Issue by jimrandomh** [LessWrong2/Lesswrong2#1211]
Benito: "30-70% of the time I upload an image, my screen breaks. This is terrible when the post I'm creating... is a string of 20 images."

**Comment by jimrandomh** [comment]
@ben-pace Could you clarify where you're uploading from, and check the console for errors?

**Comment by ben-pace** [comment]
I'm uploading from Cloudinary. I'll see if I can go re-create it.

**Comment by ben-pace** [comment]
Here's two images of it: 
<img width="1680" alt="screen shot 2018-11-01 at 15 13 47" src="https://user-images.githubusercontent.com/24915924/47860662-fa856f00-dde8-11e8-8889-8f2a9b57d126.png">
<img width="1680" alt="screen shot 2018-11-01 at 15 13 49" src="https://user-images.githubusercontent.com/24915924/47860673-007b5000-dde9-11e8-92f4-8d825c038f83.png">

**Comment by ben-pace** [comment]
I recently figured out the following fix, which is suggestive of the solution:

If I 
- create a post
- add text in the lines above and below where I want to insert an image
- save the draft
- edit the draft
- insert the imate
it works 100% of the time.

I have a sense this wasn't the case in the past. I'll bring up any other instances in the future that violate this norm or seem to.

**Comment by jimrandomh** [comment]
Could you give a procedure for reproducing the bug (as opposed to avoiding reproducing the bug)?

**Comment by Discordius** [comment]
Also, I assume by "uploading" you mean "insert the link into LW?"

**Comment by ben-pace** [comment]
@ jim

make an empty post, insert an image. this pretty reliably breaks for me.

You'll need to insert some empty lines, the editor doesn't let you insert a post on the first line, or the last, I think.

@oli yes.

**Comment by Discordius** [comment]
What is the precise thing you do to enter an image?

**Comment by Discordius** [comment]
I think the thing that causes it to crash is if you select multiple lines of text, and then try to insert an image. The intended way of inserting image is to select a bunch of text, and then press the create-image button. If you do this it should never crash

**Comment by Discordius** [comment]
In general, you should never select whitespace and try to enter an image, I can image that messing up in a bunch of ways. Obviously not a very clear UI interaction, but will at least help you not go insane.

**Comment by ben-pace** [comment]
Yeah, I think I remember in the past that selecting text has been better than not doing so.

I never select multiple lines.

**Comment by ben-pace** [comment]
The text thing works for me (just confirmed).

**Comment by ben-pace** [comment]
Images on whitespace is the standard interaction, that seems quite bad. Writers often will leave a space for an image, then insert it later (I believe).

**Comment by Discordius** [comment]
Yep, seems true. Not sure how to avoid that. The same is true for dividers. Original plan was to have a sidebar for that. I might quickly give it 3 minutes and see whether I can make that happen again.
